### PR TITLE
Beta 1 Mac changes for ModelIO, SpriteKit, MediaPlayer

### PR DESCRIPTION
**Pull request by timrisi**

### CODE REVIEW

File: src/modelio.cs [hunk 1]
```diff
@@ -113,6 +113,15 @@ interface MDLAsset : NSCopying
...
 
+		[iOS (10,3), TV (10,2), Mac (10,12,4)]
+		[Export ("objectForKeyedSubscript:")]
+		[return: NullAllowed]
+		IMDLComponent GetObject (Protocol key);
+
+		[iOS (10,3), TV (10,2), Mac (10,12,4)]
+		[Export ("setObject:forKeyedSubscript:")]
+		void SetObject ([NullAllowed] IMDLComponent obj, Protocol key);
+
```
Comment (spouliot) [inline src/modelio.cs]: Check with @VincentDondain (or his PR) IIRC there was a reason for those not being there

File: src/modelio.cs [hunk 2]
```diff
@@ -1088,6 +1097,15 @@ interface MDLObject : MDLNamed
...
 		MDLObject FromNode (SCNNode node, [NullAllowed] IMDLMeshBufferAllocator bufferAllocator);
+
+		[iOS (10,3), TV (10,2), Mac (10,12,4)]
+		[Export ("objectForKeyedSubscript:")]
+		[return: NullAllowed]
+		IMDLComponent GetObject (Protocol key);
+
+		[iOS (10,3), TV (10,2), Mac (10,12,4)]
+		[Export ("setObject:forKeyedSubscript:")]
+		void SetObject ([NullAllowed] IMDLComponent obj, Protocol key);
```
Comment (spouliot) [inline src/modelio.cs]: same

File: src/mediaplayer.cs [hunk 3]
```diff
@@ -1406,7 +1406,7 @@ interface MPNowPlayingInfoCenter {
 		[Field ("MPNowPlayingInfoPropertyIsLiveStream")]
 		NSString PropertyIsLiveStream { get; }
 
-		[iOS (10,3)]
+		[iOS (10,3)][Mac (10, 12, 3)]
```
Comment (VincentDondain) [inline src/mediaplayer.cs]: Minor no spaces between the versions for consistency (:

File: src/spritekit.cs [hunk 4]
```diff
@@ -730,13 +736,17 @@ partial interface SKSpriteNode : SKWarpable {
 		[Export ("scaleToSize:")]
 		void ScaleTo (CGSize size);
 
+		[iOS (10,0), Mac (10, 12)]
+		[Export ("attributeValues", ArgumentSemantic.Copy)]
+		NSDictionary<NSString, SKAttributeValue> AttributeValues { get; set; }
+
 #if XAMCORE_4_0
-		[iOS (9,0), Mac(10,11)]
+		[iOS (10,0), Mac(10,12)]
```
Comment (spouliot) [inline src/spritekit.cs]: We check those are runtime so, in general ours are better than Apple.
When/if updating them we need to dig up the _truth_ about the API availability. That means, minimally confirming the new values as correct.

File: src/spritekit.cs [hunk 5]
```diff
@@ -730,13 +736,17 @@ partial interface SKSpriteNode : SKWarpable {
...
 		[Export ("valueForAttributeNamed:")]
 		[return: NullAllowed]
 		SKAttributeValue GetValue (string key);
 
-		[iOS (9,0), Mac(10,11)]
+		[iOS (10,0), Mac(10,12)]
```
Comment (spouliot) [inline src/spritekit.cs]: same

File: src/spritekit.cs [hunk 6]
```diff
@@ -964,13 +974,17 @@ partial interface SKEmitterNode {
 		[Export ("particleRenderOrder", ArgumentSemantic.Assign)]
 		SKParticleRenderOrder ParticleRenderOrder { get; set; }
 
+		[iOS (10,0), Mac (10, 12)]
+		[Export ("attributeValues", ArgumentSemantic.Copy)]
+		NSDictionary<NSString, SKAttributeValue> AttributeValues { get; set; }
+
 #if XAMCORE_4_0
-		[iOS (9,0), Mac(10,11)]
+		[iOS (10,0), Mac(10,12)]
```
Comment (spouliot) [inline src/spritekit.cs]: same

File: src/spritekit.cs [hunk 7]
```diff
@@ -964,13 +974,17 @@ partial interface SKEmitterNode {
...
 		[Export ("valueForAttributeNamed:")]
 		[return: NullAllowed]
 		SKAttributeValue GetValue (string key);
 
-		[iOS (9,0), Mac(10,11)]
+		[iOS (10,0), Mac(10,12)]
```
Comment (spouliot) [inline src/spritekit.cs]: same

File: src/spritekit.cs [hunk 8]
```diff
@@ -1093,13 +1107,17 @@ partial interface SKShapeNode {
 		[Export ("lineLength")]
 		nfloat LineLength { get; }
 
+		[iOS (10,0), Mac (10, 12)]
+		[Export ("attributeValues", ArgumentSemantic.Copy)]
+		NSDictionary<NSString, SKAttributeValue> AttributeValues { get; set; }
+
 #if XAMCORE_4_0
-		[iOS (9,0), Mac(10,11)]
+		[iOS (10,0), Mac(10,12)]
```
Comment (spouliot) [inline src/spritekit.cs]: same

File: src/spritekit.cs [hunk 9]
```diff
@@ -1093,13 +1107,17 @@ partial interface SKShapeNode {
...
 		[Export ("valueForAttributeNamed:")]
 		[return: NullAllowed]
 		SKAttributeValue GetValue (string key);
 
-		[iOS (9,0), Mac(10,11)]
+		[iOS (10,0), Mac(10,12)]
```
Comment (spouliot) [inline src/spritekit.cs]: same

File: src/spritekit.cs [hunk 10]
```diff
@@ -3048,13 +3066,17 @@ interface SKTileMapNode : NSCopying, NSCoding
 		SKTileMapNode[] FromTileSet (SKTileSet tileSet, nuint columns, nuint rows, CGSize tileSize, GKNoiseMap noiseMap, NSNumber[] thresholds);
 #endif
 
+		[iOS (10,0), Mac (10, 12)]
+		[Export ("attributeValues", ArgumentSemantic.Copy)]
+		NSDictionary<NSString, SKAttributeValue> AttributeValues { get; set; }
+
 #if XAMCORE_4_0
-		[iOS (9,0), Mac(10,11)]
+		[iOS (10,0), Mac(10,12)]
```
Comment (spouliot) [inline src/spritekit.cs]: same

File: src/spritekit.cs [hunk 11]
```diff
@@ -3048,13 +3066,17 @@ interface SKTileMapNode : NSCopying, NSCoding
...
 		[Export ("valueForAttributeNamed:")]
 		[return: NullAllowed]
 		SKAttributeValue GetValue (string key);
 
-		[iOS (9,0), Mac(10,11)]
+		[iOS (10,0), Mac(10,12)]
```
Comment (spouliot) [inline src/spritekit.cs]: same

File: src/modelio.cs (hunk 1 above)
Comment (VincentDondain) [inline src/modelio.cs]: Was about to comment: see https://github.com/xamarin/xamarin-macios/pull/1601#discussion_r98778640

**Comment by VincentDondain** [comment]
After talking to @timrisi I'm gonna take over his work (thanks for the SpriteKit changes).

I'm actually gonna reuse his branch on his fork `timrisi:xcode83-MacBeta1` but I will create a new PR as I'm making the changes `SpriteKit` only.

I PRed his `MediaPlayer` changes in a separate commit already here: https://github.com/xamarin/xamarin-macios/pull/1693

`ModelIO` didn't change as explained but I added the xtro ignores here: https://github.com/xamarin/xamarin-macios/pull/1692

**Pull request closed without merging**